pkg/topology/kademlia/kademlia.go

```diff
@@ -1316,10 +1317,13 @@ func (k *Kad) Reachable(addr swarm.Address, status p2p.ReachabilityStatus) {
 }
 
 // UpdateReachability updates node reachability status.
+// The status will be updated only once.
 func (k *Kad) UpdateReachability(status p2p.ReachabilityStatus) {
-	k.logger.Infof("kademlia: updated reachability to %s", status.String())
-	k.reachability = status
-	k.metrics.ReachabilityStatus.WithLabelValues(status.String()).Set(0)
+	k.setReachabilityOnce.Do(func() {
```

Comment:
Not sure if this is what we want to do. I think it's possible that the reachability can change over the life of the node. Having a `sync.Once` seems too restrictive.

Instead, I would simply put an if loop around the metrics update call to update only if the value was different that the original.

Reply:
But then you'll have inconsistencies between what is set in metrics and what will be reported in the topology endpoint.